test/e2e/cluster/create.go

```diff
@@ -110,6 +103,27 @@ func (c *Create) Run(ctx context.Context, test TestResources) error {
 		return fmt.Errorf("loading kubeconfig: %w", err)
 	}
 
+	k8sClient, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return fmt.Errorf("creating kubernetes client: %w", err)
+	}
+
+	podIdentityAddon := addon.PodIdentityAddon{
```

Comment:
could we have a constructor for this?

I'm not against creating this "service" objects with a struct directly, but in this case it seems like to do so we have to reveal a lot of the internal details: the fact that wraps an Addon, the name of the add-on, the configuration... It appears that those details could be hidden in a constructor so they remain internal to the addon package.